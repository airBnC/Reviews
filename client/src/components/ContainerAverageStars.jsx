import React from 'react';

import TotalReviews from './TotalReviews';
import SearchReviews from './SearchReviews';
import ReviewStarsByDescriptor from './ReviewStarsByDescriptor';

const ContainerAverageStars = props => (
  <div>
    <h3> Average Stars Container </h3>
    <TotalReviews />
    <SearchReviews />
    <ReviewStarsByDescriptor />
  </div>
);

export default ContainerAverageStars;
