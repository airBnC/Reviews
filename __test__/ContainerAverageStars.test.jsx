import React from 'react';
import { configure, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import ContainerAverageStars from '../client/src/components/ContainerAverageStars';

configure({ adapter: new Adapter() });

const dummyAverageStarsObj = {
  overallAvg:2.9,
  accuracyAvg:2.85,
  communicationAvg: 2.77,
  cleanlinessAvg: 3.54,
  locationAvg: 2.62,
  checkinAvg: 3.23,
  valueAvg: 2.38,
  numReviews: 13
};

describe('ContainerAverageStars', () => {
  it('should render correctly', () => {
    const output = shallow(<ContainerAverageStars averageStarsObj={dummyAverageStarsObj} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
