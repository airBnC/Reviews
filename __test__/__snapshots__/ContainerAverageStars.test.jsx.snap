// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerAverageStars should render correctly 1`] = `
<styled.div>
  <styled.span>
    <SubcontainerAverageStarsHeader
      numReviews={13}
      overallAvg={2.9}
    />
  </styled.span>
  <ReviewStarsByDescriptor
    accuracyAvg={2.85}
    checkinAvg={3.23}
    cleanlinessAvg={3.54}
    communicationAvg={2.77}
    locationAvg={2.62}
    valueAvg={2.38}
  />
</styled.div>
`;
